src/lib/Query/Common/QueryConverter/NativeQueryConverter.php

```diff
@@ -68,7 +68,7 @@ public function convert(Query $query, array $languageSettings = [])
     {
         $params = [
             'q' => '{!lucene}' . $this->criterionVisitor->visit($query->query),
-            'fq' => '{!lucene}' . $this->criterionVisitor->visit($query->filter),
+            'fq' => '{!lucene}' . ($query->filter ? $this->criterionVisitor->visit($query->filter) : ''),
```

Comment:
I'd prefer here something more strict, given filter is null|object
```suggestion
            'fq' => '{!lucene}' . ($query->filter !== null ? $this->criterionVisitor->visit($query->filter) : ''),
```